zb-classifier.js

```diff
@@ -1987,161 +2017,177 @@ class ZigbeeClassifier {
-        sceneProperty.buttonIndex = 4;
-      }
+    for (const idx in genLevelCtrlOutputEndpoints) {
+      console.log('Processing endpoint', idx, '=',
+                  genLevelCtrlOutputEndpoints[idx]);
+      const suffix = (idx === 0) ? '' : `${idx}`;
+      const endpoint = genLevelCtrlOutputEndpoints[idx];
+      const onOffProperty = this.addButtonOnProperty(node, endpoint, suffix);
+      const levelProperty = this.addButtonLevelProperty(node, endpoint, suffix);
+
+      if (node.modelId === 'TRADFRI remote control') {
+        const firstGenOnOffOutputEndpoint = genScenesOutputEndpoints[0];
+        const genScenesOutputEndpoints =
+            node.findZhaEndpointWithOutputClusterIdHex(
+              CLUSTER_ID.GENSCENES_HEX);
+        let getScenesOutputEndpoint;
+        if (genScenesOutputEndpoints.length === 1) {
+          getScenesOutputEndpoint = genScenesOutputEndpoints[0];
+        }
+        if (!getScenesOutputEndpoint &&
+            firstGenOnOffOutputEndpoint &&
+            node.activeEndpoints[firstGenOnOffOutputEndpoint] &&
+            node.activeEndpoints[firstGenOnOffOutputEndpoint]
+              .deviceId === '0820') {
+          // Workaround: version E1810 (deviceId 0820) is missing the GENSCENES
+          // output cluster but still receives updates on that cluster
+          getScenesOutputEndpoint = firstGenOnOffOutputEndpoint;
+        }
+        if (getScenesOutputEndpoint) {
+          const sceneProperty =
+              this.addButtonSceneProperty(node, getScenesOutputEndpoint);
+          sceneProperty.buttonIndex = 4;
+        }
+
+        // This is the IKEA remote with a center button and 4 other
+        // buttons around the edge. The center button sends a toggle
+        // rather than on/off.
+
+        onOffProperty.buttonIndex = 1;
+        this.addEvents(node, {
+          '1-pressed': {
+            '@type': 'PressedEvent',
+            description: 'On button pressed and released',
+          },
+        });
 
-      // This is the IKEA remote with a center button and 4 other
-      // buttons around the edge. The center button sends a toggle
-      // rather than on/off.
+        // The remaining buttons can all generate pressed,
+        // longPressed and released events.
+        levelProperty.buttonIndex = 2;
+        const label = ['Top', 'Bottom', 'Right', 'Left'];
+        for (let i = 0; i < 4; i++) {
+          this.addEvents(node, {
+            [`${i + 2}-pressed`]: {
+              '@type': 'PressedEvent',
+              description: `${label[i]} button pressed and released`,
+            },
+            [`${i + 2}-longPressed`]: {
+              '@type': 'LongPressedEvent',
+              description: `${label[i]} button pressed and held`,
+            },
+            [`${i + 2}-released`]: {
+              '@type': 'ReleasedEvent',
+              description: `${label[i]} button released (after being held)`,
+            },
+          });
+        }
+        continue;
```

Comment:
This should remain `return` to prevent broken Tradfri remote support.

Reply:
done